api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -301,6 +304,7 @@ func (m *AzureManagedControlPlane) Validate(cli client.Client) error {
 		m.validateNetworkPluginMode,
 		m.validateDNSPrefix,
 		m.validateDisableLocalAccounts,
+		m.validateSecurityProfile,
```

Comment:
Should we also be invoking this new webhook validation for AzureManagedControlPlaneTemplate?

Reply:
okay will do it.

Reply:
/hold just in case this happens to pass CI before you get to this.